File: frontend/view/main-view.ts
import {html, LitElement, customElement} from "lit-element";
import '@vaadin/vaadin-ordered-layout/vaadin-vertical-layout';

@customElement('main-view')
export class MainView extends LitElement {
    protected createRenderRoot(): Element | ShadowRoot {
        return this;
    }

    protected render() {
        return html`
            <vaadin-vertical-layout id="main-view-root">
                <label id="text"/>
            </vaadin-vertical-layout>
        `
    }
}

File: src/main/java/ru/andreeva/shop/ui/view/MainView.java
package ru.andreeva.shop.ui.view;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.router.PageTitle;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.spring.annotation.UIScope;
import org.springframework.stereotype.Component;

@Route(value = "", layout = MainLayout.class)
@Tag("main-view")
@PageTitle("Главная")
@JsModule("./view/main-view.ts")
@UIScope
@Component
public class MainView extends LitTemplate {
    @Id("text")
    private Label testText;

    public MainView() {
        testText.setText("Добро пожаловать в АС Товароучётный менеджмент!");
    }
}
